src/odemis/gui/plugin/__init__.py

```diff
@@ -294,6 +295,7 @@ def menu_callback_wrapper(evt):
                 logging.exception("Error when processing menu entry %s of plugin %s",
                                   path[-1], self)
         main_frame.Bind(wx.EVT_MENU, menu_callback_wrapper, id=menu_item.Id)
+        self.menu_group = curr_group
```

Comment:
I tried to do that, but then it just returns None

Reply:
I totally agree that it's not a good solution. Actually, reading the code  you don't really care about the group. You care about the menu_item itself, right? As this function is decorated by `@call_in_wx_main`, it's called asynchronously, and you cannot receive its return value.
Having the menu_group or menu_item directly on the Plugin doesn't work well because this function can be called multiple times, to create multiple menu entries. 
I can think of two alternatives:
* Have a attribute on Plugin `.menu_items: Dict[str, MenuItem]` which maps the requested (full path) menu name to the actual menu item object.
* Have a optional argument to the function `pass_menu_item=False`, so that if it's True, the menu_item will be pass as argument to the callback.

I think I like a little bit more the second alternative, but I let you decide. Or do you have something else to suggest?